### PR TITLE
Support JaCoCo and other quality checks

### PR DESCRIPTION
**Pull request by rohityadavcloud**
This enables JaCoCo and other quality/history checks for the ASF Jenkins job. Also adds a workflow to generate coverage report for the PR.

### CODE REVIEW

File: .github/workflows/pull-request-checks.yml [hunk 1]
```diff
@@ -15,17 +15,45 @@
...
+      #- name: Generate JaCoCo Report
+      #  id: jacoco
+      #  uses: Madrapps/jacoco-report@v1.2
+      #  with:
+      #    paths: ${{ github.workspace }}/**/jacoco.xml
+      #    token: ${{ secrets.GITHUB_TOKEN }}
+      #    min-coverage-overall: 0
+      #    min-coverage-changed-files: 80
+      #    title: PR Code Coverage
+      #    debug-mode: true
```
Comment (DaanHoogland) [inline .github/workflows/pull-request-checks.yml:58]: we want to keep this?

File: pom.xml [hunk 2]
```diff
@@ -1152,7 +1152,6 @@
                     <version>${cs.surefire-plugin.version}</version>
                     <configuration>
                         <useSystemClassLoader>false</useSystemClassLoader>
-                        <argLine>-Djava.security.egd=file:/dev/./urandom -noverify</argLine>
```
Comment (rohityadavcloud) [inline pom.xml:1155]: Turns out this argline override, essentially disables any quality plugins that try to export their config to the mvn/cmd line. Disabling this now that linux and our build/pkgs are much faster. In my testing, I found no difference in the amount of time it took for the build.

**Review by DaanHoogland — APPROVED** [review]
looks good @rohityadavcloud . Let's see if the report makes any sense and leeds to addressable issues ;)

**Comment by rohityadavcloud** [comment]
@DaanHoogland I need some input on how the Jenkins pr-analysis job is tied to PRs. It doesn't show up now :( in the list of PR checks - https://ci-builds.apache.org/job/Cloudstack/job/cloudstack-pr-analysis/

Some more progress, now the quality checks are running and I'm able to generate clover and jacoco coverage reports. For example:
Screenshot 2022-04-07 at 5 51 15 PM

All that is required is now to enable the `-P quality` profile in the build command, for example:
```
mvn -P developer,systemvm,quality -Dsimulator -Dnoredist clean install
```

To generate the clover html report, we do for example:
```
mvn -P quality clean install clover:aggregate clover:clover
```

**Comment by DaanHoogland** [comment]
> @DaanHoogland I need some input on how the Jenkins pr-analysis job is tied to PRs. It doesn't show up now :( in the list of PR checks - [ci-builds.apache.org/job/Cloudstack/job/cloudstack-pr-analysis](https://ci-builds.apache.org/job/Cloudstack/job/cloudstack-pr-analysis/)
> 

it is there in the job configuration in jenkins, but it seems to be flakey in terms of addressing each job, @rohityadavcloud . I've never found out what made it choose to include some PRs and not others.

**Comment by rohityadavcloud** [comment]
Sadly the quality/coverage doesn't allow threaded build with -T flag, and it takes 40+mins alone to build it :( which makes using github actions not scalable for PRs:
Screenshot 2022-04-07 at 7 52 12 PM

Furthermore, it's going to be a challenge to get PR-specific coverage accurately (for ex. changes in a class which is covered by the PR's unit tests, may not fully reflect the coverage for the overall class). I'll explore other options.

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Pull request merged**